src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -308,7 +284,45 @@ private void PerformBlockPurge(IDisposable purgeRequest)
                 }
             }
 
+            if (lastMessagePurged.HasValue && OnItemPurged != null)
+            {
+                OnItemPurged(lastMessagePurged.Value);
```

Comment:
Yes.  This is mainly due perf issues.  For ordered reliable queues we shouldn't need to notify on every purged event, only on the last event of every set of events that get purged.  I suspect this will hold true for all ordered queues, but if not, we can change.